rtsp.go

```diff
@@ -390,6 +398,13 @@ func (rc *rtspCamera) initH264(session *description.Session) (err error) {
 		return fmt.Errorf("creating H264 RTP decoder: %w", err)
 	}
 
+	// Segmenter needs its own rtp decoder since the parser it is stateful
+	// and will not work concurrently with the parser used for the decode path.
+	rtpSegDec, err := f.CreateDecoder()
```

Comment:
We should be able to share the same gorstplib decoder (parser) between store and segment paths.